File: Easy/4216-weighted-word-mapping/weighted-word-mapping.cpp
class Solution{
public:
    int solve(string &s, vector<int> &wt, int m){
        int res=0;
        for(int i=0; i<s.size(); i++){
            res += wt[s[i]-97];
        }

        // cout << res << endl;

        return res;
    }

    string mapWordWeights(vector<string> &w, vector<int> &wt){
        int n=w.size();
        int m=wt.size();

        string str="";
        for(int i=0; i<n; i++){
            string s=w[i];

            int t=solve(s, wt, m);
            // cout << t%26 << " ";
            str += 'z'-(t%26);
        }

        return str;


    }
};

File: Easy/4216-weighted-word-mapping/weighted-word-mapping.py
class Solution:
    def solve(self, s, wt, m):
        res=0
        for i in range(0, len(s)):
            res += wt[ord(s[i])-ord('a')]
        
        return res
    
    def mapWordWeights(self, w: List[str], wt: List[int]) -> str:
        n=len(w)
        m=len(wt)

        str=[]
        for i in range(n):
            s=w[i]

            t=self.solve(s, wt, m)
            str.append(chr(ord('z')-(t%26)))

        return "".join(str)